docs/spock_primer.adoc

```diff
@@ -506,6 +506,21 @@ Although it is declared last, the `where` block is evaluated before the feature
 
 The `where` block is further explained in the <<data_driven_testing.adoc#data-driven-testing,Data Driven Testing>> chapter.
 
+==== Filter Blocks
```

Comment:
All alternatives I also considered.

I indeed started with `exclude` blocks as it is the more common use-case that you want to exclude some specific iterations and that it is unambiguous whether the content includes or excludes.

Until I realized that the typical usage I envision is, that the majority of iterations should be included, and just a few specific ones excluded.
As I treat the block content like `expect` block content, that is as implicit or explicit assertions, I let it evaluate and catch the exception.
Exception for flow control is evil of course, but I think here it is the right thing to do, also to stay consistent in behavior.

So the things to exclude are the things that happen more seldom and should be the ones that cause exceptions to be thrown and caught, so successful evaluation would be an included.

I considered `filter` which we already concretely talked about in the cross-compile PR and `include`, but decided for `filter` which is for example also used in Java Streams and Kotlin with the same semantics, that it defines what to include, not what to exclude, so I thought the meaning of the `filter` is widely enough known and it is better fitting, because `include` could also mean defining additional iterations like you do with `include` in GitHub action workflow matrixes.

Hence I ended up with using `filter` as that is what we do there, we filter the iterations before they are considered.